src/net_processing.cpp

```diff
@@ -2242,9 +2247,51 @@ void PeerManagerImpl::SendPings()
     for(auto& it : m_peer_map) it.second->m_ping_queued = true;
 }
 
+void PeerManagerImpl::AskPeersForTransaction(const uint256& txid, bool is_masternode)
+{
+    std::vector<PeerRef> peersToAsk;
+    peersToAsk.reserve(4);
+
+    auto maybe_add_to_nodesToAskFor = [&](const PeerRef& peer) {
+        if (peersToAsk.size() >= 4) {
+            return false;
+        }
+        if (IsInvInFilter(peer, txid)) {
+            peersToAsk.emplace_back(peer);
+        }
+        return true;
+    };
+
+    {
+        LOCK(m_peer_mutex);
+        // TODO consider prioritizing MNs again, once that flag is moved into Peer
+        for (const auto& [_, peer] : m_peer_map) {
+            if (!maybe_add_to_nodesToAskFor(peer)) {
+                break;
+            }
+        }
+    }
```

Comment:
nit: maybe simplify it?

how soon are you planning to implement this TODO?
If it's not short-term plans like very soon, it can be simplified to:

```suggestion
    {   
        LOCK(m_peer_mutex);
        // TODO consider prioritizing MNs again, once that flag is moved into Peer
        for (const auto& [_, peer] : m_peer_map) {
            if (peersToAsk.size() >= 4) {
                break; 
            }
            if (IsInvInFilter(peer, txid)) {
                peersToAsk.emplace_back(peer);
            }
        }
    }
```